modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIMUserUpdateReadOnlyTest.java

```diff
@@ -0,0 +1,117 @@
+package org.wso2.identity.integration.test.scim2.rest.api;
+
+import io.restassured.RestAssured;
+import io.restassured.response.ExtractableResponse;
+import io.restassured.response.Response;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+
+import java.rmi.RemoteException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.USERS_ENDPOINT;
+
+/**
+ * Test cases for updating users via SCIM PATCH operations in a read-only user store.
+ */
+public class SCIMUserUpdateReadOnlyTest extends SCIM2BaseTest {
+
+    private static final Log log = LogFactory.getLog(SCIMUserUpdateReadOnlyTest.class);
+
+    protected String endpointURL;
```

Comment:
Are these variables use outside the class? Otherwise lets restrict the access to private